File: companion/spss/r/b03.R
d <- read.csv("companion/spss/csv/b03.csv")
stopifnot(!anyNA(d))
print(sapply(d[c("g3", "absences")], function(v)
  c(n=length(v), mean=mean(v), sd=sd(v),
    median=median(v), min=min(v), max=max(v))))
print(t.test(d$g3, conf.level=.95)$conf.int)
print(t.test(d$absences, conf.level=.95)$conf.int)
par(mfrow=c(2, 2))
for (v in c("g3", "absences")) {
  hist(d[[v]], main=v, xlab=v)
  boxplot(d[[v]], main=v)
}
par(mfrow=c(1, 1))

File: companion/spss/python/b03.py
import pandas as pd
import numpy as np
from scipy import stats
import matplotlib.pyplot as plt

d = pd.read_csv("companion/spss/csv/b03.csv")
assert not d.isna().any().any()
print(d[["g3", "absences"]].agg(
    ["count", "mean", "std", "median", "min", "max"]))
print(stats.ttest_1samp(d.g3, 0).confidence_interval(.95))
print(stats.ttest_1samp(d.absences, 0).confidence_interval(.95))
fig, axes = plt.subplots(2, 2)
for row, v in enumerate(["g3", "absences"]):
    axes[row, 0].hist(d[v], bins=15)
    axes[row, 0].set_title(v)
    axes[row, 1].boxplot(d[v]); axes[row, 1].set_title(v)
plt.tight_layout(); plt.show()